src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -1223,6 +1235,12 @@ public void run() {
                 if (flushEventsOnClose) {
                     updateServer();
                 }
+
+                // re-persist metadata into database for good measure
+                dbHelper.insertOrReplaceKeyValue(DEVICE_ID_KEY, deviceId);
+                dbHelper.insertOrReplaceKeyValue(USER_ID_KEY, userId);
+                dbHelper.insertOrReplaceKeyLongValue(OPT_OUT_KEY, optOut ? 1L : 0L);
+                dbHelper.insertOrReplaceKeyLongValue(LAST_EVENT_TIME_KEY, lastEventTime);
```

Comment:
Persist sessionId to PREVIOUS_SESSION_ID_KEY and the other ones I mentioned above?

Reply:
persisting sessionId. The last event time gets refreshed a few lines above in the same function `refreshSessionTime`.